test/test_async.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2012-2020 Snowflake Computing Inc. All right reserved.
+#
+import time
+
+import pytest
+
+from snowflake.connector import ProgrammingError
+from snowflake.connector.constants import QueryStatus
+from snowflake.connector.errors import InterfaceError
+
+# Mark all tests in this file to time out after 2 minutes to prevent hanging forever
+pytestmark = pytest.mark.timeout(120)
+
+
+def test_async_exec(conn_cnx):
+    """Tests whether simple async query execution works.
+
+    Runs a query that takes a few seconds to finish and then totally closes connection
+    to Snowflake. Then waits enough time for that query to finish, opens a new connection
+    and fetches results. It also tests QueryStatus related functionality too.
+    """
+    with conn_cnx() as con:
+        with con.cursor() as cur:
+            cur.execute('select count(*) from table(generator(timeLimit => 5))', exec_async=True)
```

Comment:
could the method name be `execute_async` instead of having a parameter?

Reply:
Would it work if I made a convenience function on top of the current implementation?
The function `execute_async` could just call `execute` with this parameter set to true.

Reply:
Yes, that's exactly the point. Hide `exec_async` option by changing the name to `_exec_async`, though it won't be completely invisible.

Reply:
Sure, can add that!